### PR TITLE
Change the communication strategy to be chucked instead of blocked one

### PR DESCRIPTION
**Pull request by henrybarreto**

### CODE REVIEW

File: src/radio.rs [hunk 1]
```diff
@@ -8,66 +8,163 @@ use std::{
...
-    fn send_information(
+    fn send_chucked(
         tcp_connection: &TcpStream,
-        data: Vec<S>,
+        data: Vec<u8>,
     ) -> Result<bool, Box<dyn Error>> {
-        let tcp_stream_ref: Rc<TcpStream> = Rc::new(tcp_connection.try_clone()?);
+        let tcp_connection_ref: Rc<TcpStream> = Rc::new(tcp_connection.try_clone()?);
+
+        println!("DATA TO SEND: {:?}", data);
```
Comment (henrybarreto) [inline src/radio.rs:32]: The data is sent here...

File: src/radio.rs [hunk 2]
```diff
@@ -8,66 +8,163 @@ use std::{
...
+            }
+            // Receive data
+            let mut data_received= Self::receive_bytes(128, tcp_connection_ref.try_clone()?)?;
+            println!("RECEIVED DATA CHUCK: {:?}", data_received);
+            data.append(&mut data_received);
+            // Send a Okay response
+            Self::send_bytes(&[1], tcp_connection_ref.try_clone()?)?;
+        };
+
+        println!("RECEIVED DATA: {:?}", data);
```
Comment (henrybarreto) [inline src/radio.rs:121]: But when the other get the bytes, the total data differs only per 8 bytes in front of the original sent data.

**Comment by henrybarreto** [comment]
I have no idea what is happening...

File: src/radio.rs [hunk 3]
```diff
@@ -8,64 +8,94 @@ use std::{
...
         tcp_stream.write(&buffer)
     }
-    fn recv_data(size: usize, mut tcp_stream: TcpStream) -> Result<Vec<u8>, std::io::Error> {
+    fn receive_bytes(size: usize, mut tcp_stream: TcpStream) -> Result<Vec<u8>, std::io::Error> {
         let mut buffer = vec![0; size];
         match tcp_stream.read(&mut buffer) {
-            Ok(_) => Ok(buffer),
+            Ok(_) => {
+                Ok(buffer)
+            }
```
Comment (otavio) [inline src/radio.rs]:
```rust
tcp_stream.read(&mut buffer).map(|_| buffer)?
```

I believe this replace this ugly match.

File: src/radio.rs [hunk 4]
```diff
@@ -8,64 +8,94 @@ use std::{
...
+        for chunk in chunks {
+            Self::send_bytes(chunk, tcp_connection_ref.try_clone()?)?;
+            let okay = Self::receive_bytes(1, tcp_connection_ref.try_clone()?)?;
+        }
+        Self::send_bytes(remainder, tcp_connection_ref.try_clone()?)?;
+        let okay = Self::receive_bytes(1, tcp_connection_ref.try_clone()?)?;
+
+        // I know, this is not for real
+        Self::send_bytes(&bincode::serialize(&-1)?, tcp_connection_ref.try_clone()?)?;
+        let okay = Self::receive_bytes(1, tcp_connection_ref.try_clone()?)?;
```
Comment (otavio) [inline src/radio.rs]: `okay` is unused.

**Review by otavio — CHANGES_REQUESTED** [review]

**Comment by nightlark** [comment]
@henrybarreto not sure how you're using the `invalid` label, but you might want to avoid it except for PRs that are actually spam for the duration of hacktoberfest, since it seems like it could be counted towards bans.

**Comment by henrybarreto** [comment]
> @henrybarreto not sure how you're using the `invalid` label, but you might want to avoid it except for PRs that are actually spam for the duration of hacktoberfest, since it seems like it could be counted towards bans.

Thanks! I will avoid this one when the event begins.

**Pull request merged**